File: 链蒸蒸简单/出现最频繁的偶数元素.java
package 链蒸蒸简单;

import java.util.HashMap;

public class 出现最频繁的偶数元素 {
    public int mostFrequentEven(int[] nums) {
        int res = -1;
        int maxfre = 0;
        var cnt = new HashMap<Integer, Integer>();
        for (int num : nums) {
            if (num % 2 ==1) continue;
            cnt.put(num, cnt.getOrDefault(num, 0) + 1);
            if (cnt.get(num) == maxfre && num< res || cnt.get(num) >maxfre) {
                res = num;
                maxfre = cnt.get(num);
            }
        }
        return res;
    }
}

File: 链蒸蒸简单/ 出现最频繁的偶数元素.c
#include <stdio.h>
#include <string.h>

int mostFrequentEven(int *nums, int numsSize) {
    int cnt[100005];
    //memset(cnt, 0, sizeof(int) * 100005);
    int res = -1;
    int maxfre = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] % 2 == 1) continue;
        cnt[nums[i]]++;
    }
    for (int i = 0; i < 100005; i++) {
        if (cnt[i] > maxfre) {
            maxfre = cnt[i];
            res = i;
        }
    }
    return res;
}


int main() {
    int nums[] = {2, 2, 2, 4, 4, 4};
    int a = mostFrequentEven(nums, 6);
    printf("%d", a);
}